Sources/TestingMacros/Support/DiagnosticMessage+Diagnosing.swift

```diff
@@ -36,58 +36,171 @@ func diagnoseIssuesWithTags(in traitExprs: [ExprSyntax], addedTo attribute: Attr
+      func checkForValidDeclReferenceExpr(_ declReferenceExpr: DeclReferenceExprSyntax) {
+        // This is the name of a type or symbol. If there are argument names
+        // (unexpected in this context), it's a function reference and is
+        // unsupported.
+        if declReferenceExpr.argumentNames != nil {
+          context.diagnose(.tagExprNotSupported(tagExpr, in: attribute))
+        }
+      }
+      func checkForValidBaseExpr(_ baseExpr: ExprSyntax) {
+        if let baseExpr = baseExpr.as(MemberAccessExprSyntax.self) {
+          checkForValidDeclReferenceExpr(baseExpr.declName)
+          if let baseBaseExpr = baseExpr.base {
+            checkForValidBaseExpr(baseBaseExpr)
           }
+        } else if let baseExpr = baseExpr.as(DeclReferenceExprSyntax.self) {
+          checkForValidDeclReferenceExpr(baseExpr)
         } else {
-          // This tag is not of a supported expression type.
+          // The base expression was some other kind of expression and is
+          // not supported.
           context.diagnose(.tagExprNotSupported(tagExpr, in: attribute))
         }
       }
-    default:
-      // This is not a tag list (as far as we know.)
+      if let baseExpr = tagExpr.base {
+        checkForValidBaseExpr(baseExpr)
+      }
+    } else {
+      // This tag is not of a supported expression type.
+      context.diagnose(.tagExprNotSupported(tagExpr, in: attribute))
+    }
+  }
+}
+
+/// Diagnose issues with a `.bug()` trait in a parsed attribute.
+///
+/// - Parameters:
+///   - traitExpr: The `.bug()` expression.
+///   - attribute: The `@Test` or `@Suite` attribute.
+///   - context: The macro context in which the expression is being parsed.
+private func _diagnoseIssuesWithBugTrait(_ traitExpr: FunctionCallExprSyntax, addedTo attribute: AttributeSyntax, in context: some MacroExpansionContext) {
+  // If the first argument to the .bug() trait is unlabelled and a string
+  // literal, check that it can be parsed as a URL or at least as an integer.
+  guard let arg = traitExpr.arguments.first.map(Argument.init),
+        arg.label == nil,
+        let stringLiteralExpr = arg.expression.as(StringLiteralExprSyntax.self),
+        let urlString = stringLiteralExpr.representedLiteralValue else {
+    return
+  }
+
+  if UInt64(urlString) != nil {
+    // The entire URL string can be parsed as an integer, so allow it. Although
+    // the testing library prefers valid URLs here, some bug-tracking systems
+    // might not provide URLs, or might provide excessively long URLs, so we
+    // allow numeric identifiers as a fallback.
+    return
+  }
+
+  func isURLStringValid(_ urlString: String) -> Bool {
+    guard urlString.allSatisfy(\.isASCII),
+          let colonIndex = urlString.firstIndex(of: ":") else {
+      // This can't be a valid URL as far as we're concerned. Exit early without
+      // properly parsing it.
+      return false
+    }
+
+#if SWT_TARGET_OS_APPLE || os(Linux)
```

Comment:
This might be good to replicate into the C side.

Reply:
We already use include guards for various headers, so it isn't an issue.